ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -169,6 +192,24 @@ public void execute(Input input, ActionListener<Output> listener) {
 
     }
 
+    /**
+     * Get question from existing interaction
+     * @param memoryId conversation id
+     * @param interactionId interaction id
+     * @param inputDataSet input parameters
+     */
+    private void getQuestionFromInteraction(String memoryId, String interactionId, RemoteInferenceInputDataSet inputDataSet) {
+        try {
+            PlainActionFuture<GetInteractionResponse> future = PlainActionFuture.newFuture();
+            client.execute(GetInteractionAction.INSTANCE, new GetInteractionRequest(memoryId, interactionId), future);
+
+            GetInteractionResponse interactionResponse = future.get();
```

Comment:
I don't think we should have any blocking invocation in an opensearch thread. Can you change to non-blocking way like listener?

Reply:
sure, changed it to non-blocking way.